tests/subsys/logging/log_api/src/test.inc

```diff
@@ -654,17 +649,41 @@ void test_cc(void)
 #if __cplusplus
 	PRINT("\t C++: yes\n");
 #endif
+	return NULL;
+}
 
-	ztest_test_suite(test_log_api,
-			 ztest_unit_test(test_log_various_messages),
-			 ztest_unit_test(test_log_backend_runtime_filtering),
-			 ztest_unit_test(test_log_overflow),
-			 ztest_unit_test(test_log_arguments),
-			 ztest_unit_test(test_log_from_declared_module),
-			 ztest_unit_test(test_log_msg_dropped_notification),
-			 ztest_unit_test(test_log_panic)
-			);
-	ztest_run_test_suite(test_log_api);
-
-	log_api_suite_teardown();
+static void log_api_suite_before(void *data)
+{
+	ARG_UNUSED(data);
+	while(LOG_PROCESS()) {
+	}
 }
+
+/*test case main entry*/
+#define WRAP_TEST(test_name, suffix)                       \
+	ZTEST(test_log_api_##suffix, test_name##_##suffix) \
+	{                                                  \
+		test_name();                               \
+	}
+
+#if __cplusplus
```

Comment:
it would be less repetitions if only suffix name would be defined.
```
#if __cplusplus
#define TEST_SUFFIX cxx
#else 
#define TEST_SUFFIX cc
#endif

Reply:
Done